gamestonk_terminal/portfolio_optimization/port_opt_api.py

```diff
@@ -0,0 +1,360 @@
+        type=float,
+        dest="value",
+        help="Amount to allocate to portfolio",
+    )
+    parser.add_argument(
+        "--pie",
+        action="store_true",
+        dest="pie",
+        default=False,
+        help="Display a pie chart for weights",
+    )
+    weights = {}
+    prop = {}
+    prop_sum = 0
+
+    try:
+        ns_parser = parse_known_args_and_warn(parser, other_args)
+        if not ns_parser:
+            return
+        for stock in list_of_stocks:
+            stock_prop = yf.Ticker(stock).info[property_type]
+            if stock_prop is None:
+                stock_prop = 0
+            prop[stock] = stock_prop
+            prop_sum += stock_prop
+        for k, v in prop.items():
+            weights[k] = round(v / prop_sum, 5) * ns_parser.value
+        if ns_parser.pie:
+            pie_chart_weights(weights)
+        return weights
+
+    except Exception as e:
+        print(e)
+        print("")
+        return
+
+
+def show_ef(list_of_stocks: List[str], other_args: List[str]):
+    parser = argparse.ArgumentParser(add_help=False, prog="ef")
+
+    parser.add_argument(
+        "-p",
+        "--period",
+        default="3mo",
+        dest="period",
+        help="period to get yfinance data from",
+        choices=period_choices,
+    )
+    parser.add_argument(
+        "-n", default=300, dest="n_port", help="number of portfolios to simulate"
+    )
+
+    try:
+        ns_parser = parse_known_args_and_warn(parser, other_args)
+        if not ns_parser:
+            return {}
+        stock_prices = process_stocks(list_of_stocks, ns_parser.period)
+        mu = expected_returns.mean_historical_return(stock_prices)
+        S = risk_models.sample_cov(stock_prices)
+        ef = EfficientFrontier(mu, S)
+        fig, ax = plt.subplots(figsize=plot_autoscale(), dpi=PLOT_DPI)
+
+        # Generate random portfolios
+        n_samples = ns_parser.n_port
+        w = np.random.dirichlet(np.ones(len(mu)), n_samples)
+        rets = w.dot(mu)
+        stds = np.sqrt(np.diag(w @ S @ w.T))
+        sharpes = rets / stds
+        ax.scatter(stds, rets, marker=".", c=sharpes, cmap="viridis_r")
+
+        plotting.plot_efficient_frontier(ef, ax=ax, show_assets=True)
+        # Find the tangency portfolio
+        ef.max_sharpe()
+        ret_sharpe, std_sharpe, _ = ef.portfolio_performance()
+        ax.scatter(std_sharpe, ret_sharpe, marker="*", s=100, c="r", label="Max Sharpe")
+
+        ax.set_title("Efficient Frontier")
+        ax.legend()
+        plt.tight_layout()
+        plt.grid(b=True, which="major", color="#666666", linestyle="-")
+        plt.minorticks_on()
+        plt.grid(b=True, which="minor", color="#999999", linestyle="-", alpha=0.2)
+
+        if gtff.USE_ION:
+            plt.ion()
+
+        plt.show()
+        print("")
+
+    except Exception as e:
+        print(e)
+        print("")
+
+
+def ef_portfolio(list_of_stocks: List[str], port_type: str, other_args: List[str]):
```

Comment:
You don't need to specify list_of_stocks, you can use just `stocks`. The typing properties will tell you the type of the variable